app/backend/src/couchers/servicers/auth.py

```diff
@@ -413,12 +418,31 @@ def CompletePasswordReset(self, request, context):
             else:
                 context.abort(grpc.StatusCode.NOT_FOUND, errors.INVALID_TOKEN)
 
-    def CompleteChangeEmail(self, request, context):
-        """
-        Completes an email change request.
+    # Only called if user doesn't have password
+    def ConfirmChangeEmailWithOldAddress(self, request, context):
+        with session_scope() as session:
+            user = (
+                session.query(User)
+                .filter(User.old_email_token == request.change_email_token)
+                .filter(User.old_email_token_created <= func.now())
+                .filter(User.old_email_token_expiry >= func.now())
+                .one_or_none()
+            )
+            if user:
```

Comment:
abort with a shortcut (`if not user: ...abort`) instead of doing the main logic in the `if` branch